webserver/static/js/listens-viewer.js

```diff
@@ -17,18 +18,42 @@ var ListensViewer = React.createClass({
                           "in the data-user-id property.");
             return;
         }
-        $.get("/listen/user/" + container.dataset.userId, function(data) {
-            if (this.isMounted()) {
-                this.setState({
-                    userId: container.dataset.userId,
-                    listens: data.payload.listens
-                });
-            }
-        }.bind(this));
+        window.addEventListener('popstate', this.handleLoadData);
+        this.handleLoadData();
     },
     componentDidUpdate: function() {
         jQuery("abbr.timeago").timeago();
     },
+    handleLoadData: function () {
+        if (this.state.ignorePopState) { return; }
```

Comment:
You may not need `ignorePopState` at all. If you just want to ignore events that get triggered by `pushState`, but not ones triggered by the user, then https://developer.mozilla.org/en-US/docs/Web/Events/popstate says:

> If the history entry being activated was created by a call to history.pushState() or was affected by a call to history.replaceState(), the popstate event's `state` property contains a copy of the history entry's state object.

So, make sure this function accepts `event` as the first argument, and do `if (event.state) { return; }` instead? (Probably with a comment to explain why it's doing that.)